src/main/java/hudson/plugins/openid/OpenIdSsoSecurityRealm.java

```diff
@@ -160,12 +160,16 @@ public Authentication authenticate(Authentication authentication) throws Authent
     /**
      * The login process starts from here.
      */
-    public HttpResponse doCommenceLogin(@QueryParameter String from) throws IOException, OpenIDException {
+    public HttpResponse doCommenceLogin(@QueryParameter String from) throws IOException, OpenIDException, IllegalStateException {
```

Comment:
Good practice (IMHO): Add <code>@Restricted(NoExternalUse)</code> to Stapler methods when you have a chance